pkg/local_workflows/whoami_workflow.go

```diff
@@ -0,0 +1,142 @@
+package localworkflows
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/snyk/go-application-framework/internal/api/contract"
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+	"github.com/spf13/pflag"
+)
+
+const (
+	workflowName     = "whoami"
+	mimeTypeJSON     = "application/json"
+	endpoint         = "/user/me"
+	apiVersion       = "/v1"
+	experimentalFlag = "experimental"
+	jsonFlag         = "json"
+)
+
+// define a new workflow identifier for this workflow
+var WORKFLOWID_WHOAMI workflow.Identifier = workflow.NewWorkflowIdentifier(workflowName)
+
+// InitWhoAmIWorkflow initialises the whoAmI workflow before registering it with the engine.
+func InitWhoAmIWorkflow(engine workflow.Engine) error {
+	// initialise workflow configuration
+	whoAmIConfig := pflag.NewFlagSet(workflowName, pflag.ExitOnError)
+	// add experimental flag to configuration
+	whoAmIConfig.Bool(experimentalFlag, false, "enable experimental whoAmI command")
+	// add json flag to configuration
+	whoAmIConfig.Bool(jsonFlag, false, "output in json format")
+
+	// register workflow with engine
+	_, err := engine.Register(WORKFLOWID_WHOAMI, workflow.ConfigurationOptionsFromFlagset(whoAmIConfig), whoAmIWorkflowEntryPoint)
+	return err
+}
+
+// whoAmIWorkflowEntryPoint is the entry point for the whoAmI workflow.
+// it calls the `/user/me` endpoint and returns the authenticated user's username
+// it can optionally return the full `/user/me` payload response if the json flag is set
+func whoAmIWorkflowEntryPoint(invocationCtx workflow.InvocationContext, _ []workflow.Data) (output []workflow.Data, err error) {
+	// get necessary objects from invocation context
+	config := invocationCtx.GetConfiguration()
+	logger := invocationCtx.GetLogger()
+	httpClient := invocationCtx.GetNetworkAccess().GetHttpClient()
+
+	logger.Println("whoAmI workflow start")
+
+	// only run if experimental flag is set
+	if !config.GetBool(experimentalFlag) {
+		return nil, fmt.Errorf("set `--experimental` flag to enable whoAmI command")
+	}
+
+	// define userme API endpoint
+	baseUrl := config.GetString(configuration.API_URL)
+	url := baseUrl + apiVersion + endpoint
+
+	// call userme API endpoint
+	userMe, err := fetchUserMe(httpClient, url, logger)
+	if err != nil {
+		return nil, fmt.Errorf("error while fetching user: %w", err)
+	}
+
+	// extract user from response
+	user, err := extractUser(userMe, logger)
+	if err != nil {
+		return nil, fmt.Errorf("error while extracting user: %w", err)
+	}
+
+	// return full payload if json flag is set
+	if config.GetBool(jsonFlag) {
+		// parse response
+		userMeData := createWorkflowData(userMe)
+
+		// return userme data
+		return []workflow.Data{userMeData}, err
+	}
+
+	userData := createWorkflowData(user)
```

Comment:
In this case the mimetype should be `plain/text` rather than `application/json`.

Reply:
good catch, fixed :)